app/controllers/attributes_controller.rb

```diff
@@ -1,28 +1,20 @@
 class AttributesController < ApplicationController
-  before_action :fetch_govuk_account_session
+  before_action :require_govuk_account_session!
 
   def show
     attribute_names = params.fetch(:attributes)
 
     return unless validate_attributes(attribute_names)
 
-    values = attribute_names.each_with_object({}) do |name, values_hash|
-      if user_attributes.stored_locally? name
-        raise NotImplementedError
-      else
-        oauth_response = OidcClient.new.get_attribute(
-          attribute: name,
-          access_token: @govuk_account_session[:access_token],
-          refresh_token: @govuk_account_session[:refresh_token],
-        )
-        @govuk_account_session[:access_token] = oauth_response[:access_token]
-        @govuk_account_session[:refresh_token] = oauth_response[:refresh_token]
-        values_hash[name] = oauth_response[:result]
-      end
-    end
+    local_attributes = attribute_names.select { |name| user_attributes.stored_locally? name }
+    remote_attributes = attribute_names.reject { |name| user_attributes.stored_locally? name }
+
+    raise NotImplementedError if local_attributes.any?
```

Comment:
Both in `show` and `update` we fetch the attributes, validate them, separate them in local and remote, raise an error if we have local ones. I wonder if we can extract some of this logic from the two actions into private methods.

Reply:
How's 1ac9f362be1876108fdea1b466dd2e3f6f846cc5?

Reply:
Looks good, the only thing I don't understand is why `get_attributes_from_params` returns `[local_attributes, remote_attributes]` instead of just `remote_attributes`. We already know there won't be any at that point because we have `raise NotImplementedError if local_attributes.any?` above.

Returning only `remote_attributes`, we could have `remote_attributes = get_attributes_from_params(...)` instead of `_local_attributes, remote_attributes = get_attributes_from_params(...)`. Or am I missing something?

Reply:
Ok, I'll change it to just return `remote_attributes`.  We will have `local_attributes` at some point, but not yet.